cmd/operator/api/v1alpha1/kubearchiveconfig_webhook.go

```diff
@@ -13,65 +14,86 @@ import (
+	}
 	kubearchiveconfiglog.Info("default", "namespace", kac.Namespace, "name", kac.Name)
+	return nil
 }
 
 //+kubebuilder:webhook:path=/validate-kubearchive-kubearchive-org-v1alpha1-kubearchiveconfig,mutating=false,failurePolicy=fail,sideEffects=None,groups=kubearchive.kubearchive.org,resources=kubearchiveconfigs,verbs=create;update,versions=v1alpha1,name=vkubearchiveconfig.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Validator = &KubeArchiveConfig{}
+type KubeArchiveConfigCustomValidator struct {
+	kubearchiveResourceName string
+}
 
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (kac *KubeArchiveConfig) ValidateCreate() (admission.Warnings, error) {
-	kubearchiveconfiglog.Info("validate create", "namespace", kac.Namespace, "name", kac.Name)
+var _ webhook.CustomValidator = &KubeArchiveConfigCustomValidator{}
 
-	if kac.Name != kubearchiveResourceName {
-		return nil, fmt.Errorf("The KubeArchiveConfig resource must be named '%s'.", kubearchiveResourceName)
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (kaccv *KubeArchiveConfigCustomValidator) ValidateCreate(
+	_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	kac, ok := obj.(*KubeArchiveConfig)
+	if !ok {
+		return nil, fmt.Errorf("expected an KubeArchiveConfig object but got %T", obj)
 	}
+	kubearchiveconfiglog.Info("validate create", "namespace", kac.Namespace, "name", kac.Name)
 
-	return kac.validateCELExpressions()
+	return kaccv.validateKAC(kac)
 }
 
-// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
-func (kac *KubeArchiveConfig) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
-	kubearchiveconfiglog.Info("validate update", "namespace", kac.Namespace, "name", kac.Name)
-
-	if kac.Name != kubearchiveResourceName {
-		return nil, fmt.Errorf("Cannot update KubeArchiveConfig resource named '%s'.", kubearchiveResourceName)
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type
+func (kaccv *KubeArchiveConfigCustomValidator) ValidateUpdate(
+	_ context.Context, _ runtime.Object, new runtime.Object) (admission.Warnings, error) {
+	kac, ok := new.(*KubeArchiveConfig)
+	if !ok {
+		return nil, fmt.Errorf("expected an KubeArchiveConfig object but got %T", new)
 	}
+	kubearchiveconfiglog.Info("validate update", "namespace", kac.Namespace, "name", kac.Name)
 
-	return kac.validateCELExpressions()
+	return kaccv.validateKAC(kac)
 }
 
-// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
-func (kac *KubeArchiveConfig) ValidateDelete() (admission.Warnings, error) {
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type
+func (kaccv *KubeArchiveConfigCustomValidator) ValidateDelete(
+	_ context.Context, new runtime.Object) (admission.Warnings, error) {
```

Comment:
This is a nit, but I think this would be easier to read if the function declaration can't fit on one line

```suggestion
func (kaccv *KubeArchiveConfigCustomValidator) ValidateDelete(
	_ context.Context, new runtime.Object,
) (admission.Warnings, error) {
```

Reply:
agree! fixed